### ISSUE TITLE
Cache Stats

### ISSUE THREAD

**Issue by jbhamilton** [PHPSocialNetwork/phpfastcache#272]
I'm providing a GUI interface around cached data and I'm using `driver_stats()` method to get a summary of all the data in the cache, but it is unfortunately not including the expired time stamp as a key. Right now it only provides the size (bytes) and the write time. I'm I missing something simple I could do to get this or should I create a fork?

**Comment by khoaofgod** [comment]
which driver you planned to use? and can you provide a format that you would like us to make for driver_stats ? 
The problems with the stats is each driver provider different return back base on their built-in functions. Let's find a universal format for all of them.

**Comment by jbhamilton** [comment]
For my needs I'm only using the files driver. I understand that it would need to be implemented across all drivers uniformly and right now it appears each driver stats returns a consistent root level structure:
```
        array(
          'info' => '',
          'size' => '',
          'data' => '',
        );
```
I have not checked, and I'm assuming that the structure of `data` varies between the drivers. 

Ultimately I would like to see `data` return a consistent structure across drivers like:
```
'data' => Array(
    'key' => Array(
         'size' => 203943
         'write_time' => 1461780990,
         'expire_time' => 1461781190
    ),
    ...
)
```

Also the data stored in `info` should be made use friendly consistent keys and should not include `[bytes]` as a string, as it should be specified in the documentation that `driver_stats()` returns all size values as bytes.

So this:
```
        $res[ 'info' ] = array(
          'Total [bytes]' => $total,
          'Expired and removed [bytes]' => $removed,
          'Current [bytes]' => $res[ 'size' ],
        );
```
Should look more like this:
```
        $res[ 'info' ] = array(
          'total_size' => $total,
          'expired_and_removed_size' => $removed,
          'current_size' => $res[ 'size' ],
        );
```

**Comment by khoaofgod** [comment]
I see. I put this one in progress, and it will be done in 5 - 7 days for files, sqlite and memcached first.

**Comment by jbhamilton** [comment]
Awesome! If theres anything I can do I may have a few hours I could spare to help. Let me know.

**Comment by Geolim4** [comment]
For the v4 we will keep this as an array, anyway for the v4 an object *CacheStatistic* (with a cool API doc) will be returned.
But keep in mind that this is an enhancement and not a bugfix, so bug will be fixed first :)

**Comment by khoaofgod** [comment]
@jbhamilton if you can do this for Files Cache, can you help me with Files ? then just commit it, I will just duplicated ur code for other driver.

**Comment by Geolim4** [comment]
For your information the v5 has implemented a specific object for the statistic: 5d545e49b6c23d55d7273d0074b9cd26d368d2fa

The documentation about this will follow when the v5 come out :)

**Comment by Geolim4** [comment]
An objet has been dedicated to this work in V5, see:
https://github.com/PHPSocialNetwork/phpfastcache/wiki/[V5]-The-cache-statistics
Unfortunately we can't extract statistic key-based due to performances issues.
